src/Hosting/Hosting/src/GenericHost/GenericWebHostService.cs

```diff
@@ -158,6 +159,12 @@ static string ExpandPorts(string ports, string scheme)
         await Server.StartAsync(httpApplication, cancellationToken);
         HostingEventSource.Log.ServerReady();
 
+        // See HostingEnvironmentExtensions.Initialize
+        if (HostingEnvironment.WebRootFileProvider is NullFileProvider)
+        {
+            Log.WebRootPathNotFound(LifetimeLogger, Path.GetFullPath(Path.Combine(HostingEnvironment.ContentRootPath, HostingEnvironment.WebRootPath ?? "wwwroot")));
+        }
```

Comment:
Doesn't this always end up logging if you don't have a web root? We don't want that.

Reply:
Moved to the StaticFileMiddleware. Now it only logs if you don't have webroot but do have UseStaticFile.